packages/common-scripts/src/deploy.ts

```diff
@@ -0,0 +1,415 @@
+import { RPC } from "@ckb-lumos/rpc";
+import { Set } from "immutable";
+const { ScriptValue } = values;
+
+function bytesToHex(bytes: Uint8Array): string {
+  return `0x${[...bytes].map((b) => b.toString(16).padStart(2, "0")).join("")}`;
+}
+
+async function findCellsByLock(
+  lockScript: Script,
+  cellProvider: CellProvider
+): Promise<Cell[]> {
+  const collector = cellProvider.collector({ lock: lockScript, type: "empty" });
+  const cells: Cell[] = [];
+  for await (const cell of collector.collect()) {
+    cells.push(cell);
+  }
+  return cells;
+}
+
+function updateOutputs(
+  txSkeleton: TransactionSkeletonType,
+  output: Cell
+): TransactionSkeletonType {
+  const cellCapacity = minimalCellCapacity(output);
+  output.cell_output.capacity = `0x${cellCapacity.toString(16)}`;
+  txSkeleton = txSkeleton.update("outputs", (outputs) => {
+    return outputs.push(output);
+  });
+
+  return txSkeleton;
+}
+
+function updateCellDeps(
+  txSkeleton: TransactionSkeletonType,
+  config?: Config
+): TransactionSkeletonType {
+  txSkeleton = txSkeleton.update("cellDeps", (cellDeps) => {
+    return cellDeps.clear();
+  });
+  config = config || getConfig();
+  const secp256k1Config = config.SCRIPTS.SECP256K1_BLAKE160;
+  const secp256k1MultiSigConfig = config.SCRIPTS.SECP256K1_BLAKE160_MULTISIG;
+  if (!secp256k1Config || !secp256k1MultiSigConfig) {
+    throw new Error(
+      "Provided config does not have SECP256K1_BLAKE160 or SECP256K1_BLAKE160_MULTISIG script setup!"
+    );
+  }
+  txSkeleton = txSkeleton.update("cellDeps", (cellDeps) => {
+    return cellDeps.push(
+      {
+        out_point: {
+          tx_hash: secp256k1Config.TX_HASH,
+          index: secp256k1Config.INDEX,
+        },
+        dep_type: secp256k1Config.DEP_TYPE,
+      },
+      // TODO: optimize me, push dep directly without checking actual locks used would cause bigger tx
+      {
+        out_point: {
+          tx_hash: secp256k1MultiSigConfig.TX_HASH,
+          index: secp256k1MultiSigConfig.INDEX,
+        },
+        dep_type: secp256k1MultiSigConfig.DEP_TYPE,
+      }
+    );
+  });
+
+  return txSkeleton;
+}
+
+async function completeTx(
+  txSkeleton: TransactionSkeletonType,
+  fromAddress: string,
+  config?: Config
+): Promise<TransactionSkeletonType> {
+  const inputCapacity = txSkeleton
+    .get("inputs")
+    .map((c) => BigInt(c.cell_output.capacity))
+    .reduce((a, b) => a + b, BigInt(0));
+  const outputCapacity = txSkeleton
+    .get("outputs")
+    .map((c) => BigInt(c.cell_output.capacity))
+    .reduce((a, b) => a + b, BigInt(0));
+  const needCapacity = outputCapacity - inputCapacity;
+  txSkeleton = await injectCapacity(txSkeleton, fromAddress, needCapacity, {
+    config: config,
+  });
+  return txSkeleton;
+}
+
+async function injectCapacity(
+  txSkeleton: TransactionSkeletonType,
+  fromAddress: string,
+  amount: bigint,
+  { config = undefined }: Options = {}
+): Promise<TransactionSkeletonType> {
```

Comment:
since we should support the SECP256K1 and SECP256K1_MULTISIG, i found that the implementation of `injectCapacity` here seems very similar to secp256k1's `transfer`, which means that the `injectCapacity` only support `SECP256K1_BLAKE160` but not `MULTI_SIG`, can we call the functions provided by secp256k1/multi_sig directly to implement the `injectCapacity` here?